plugins/pastetools/filter/image.js

```diff
@@ -300,13 +362,60 @@
 	}
 
 	function createSrcWithBase64( img ) {
-		var isSupportedType = CKEDITOR.tools.array.indexOf( CKEDITOR.pasteFilters.image.supportedImageTypes, img.type ) !== -1;
+		var isSupportedType = CKEDITOR.tools.array.indexOf( CKEDITOR.pasteFilters.image.supportedImageTypes, img.type ) !== -1,
+			data = img.hex;
 
 		if ( !isSupportedType ) {
 			return null;
 		}
 
-		return img.type ? 'data:' + img.type + ';base64,' + CKEDITOR.tools.convertBytesToBase64( CKEDITOR.tools.convertHexStringToBytes( img.hex ) ) : null;
+		if ( typeof data === 'string' )  {
+			data = CKEDITOR.tools.convertHexStringToBytes( img.hex );
+		}
+
+		return img.type ? 'data:' + img.type + ';base64,' + CKEDITOR.tools.convertBytesToBase64( data ) : null;
+	}
+
+	function convertBlobUrlToBase64( blobUrlSrc ) {
+		return new CKEDITOR.tools.promise( function( resolve ) {
+			CKEDITOR.ajax.load( blobUrlSrc, function( arrayBuffer ) {
+				var data = new Uint8Array( arrayBuffer ),
+					imageType = getImageTypeFromHeader( data.subarray( 0, 4 ) ),
+					base64 = createSrcWithBase64( {
+						type: imageType,
+						hex: data
+					} );
+
+				resolve( base64 );
+			} , 'arraybuffer' );
+		} );
+	}
+
+	function getImageTypeFromHeader( bytesArray ) {
+		var header = '',
+			fileType = null,
+			bytesHeader = bytesArray.subarray( 0, 4 );
+
+		for ( var i = 0; i < bytesHeader.length; i++ ) {
+			header += bytesHeader[ i ].toString( 16 );
+		}
+
+		switch ( header ) {
```

Comment:
That's some dark magic 😄 I'm not sure if it should be named `header`, probably [File signature](https://en.wikipedia.org/wiki/File_signature) is more accurate here. It would be good to have real, existing images with these signatures, to confirm that the feature covers all expected image types. These can be checked e.g. here: https://www.filesignatures.net/index.php?search=FFD8FFE0&mode=SIG although I suppose it may be hard to get all `spf`, `jfif` etc extension examples and I'm not sure if Word is not converting them also into a more unified format.